.github/workflows/create-release.yaml

```
@@ -1,53 +1,280 @@
+
+      - name: Validate branch
+        run: |
+          #parse version into major, minor, patch
+          major=$(echo ${{ inputs.name }} | cut -d. -f1)
+          minor=$(echo ${{ inputs.name }} | cut -d. -f2)
+          patch=$(echo ${{ inputs.name }} | cut -d. -f3)
+
+          if [ "${{ github.ref_name }}" = "main" ]; then
+            if [ $patch -ne 0 ]; then
+              echo "For patch releases, run the workflow from the release branch."
+              exit 1
+            else
+              git checkout -b release-$major.$minor origin/main
+              git push origin release-$major.$minor
+            fi
+          elif [[ "${{ github.ref_name }}" == release-* ]]; then
+
+            branch_version=$(echo "${{ github.ref_name }}" | sed 's/release-//')
+            if [ "$branch_version" != "$major.$minor" ]; then
+              echo "Release branch ${{ github.ref_name }} does not match tag version."
+              exit 1
+            fi
+            if [ $patch -eq 0 ]; then
+              echo "For major or minor releases, run the workflow from the main branch."
+              exit 1
+            fi
+          else
+            echo "Invalid branch ${{ github.ref_name }}."
+            exit 1
+          fi
+
+      - name: Create lightweight tag and push
+        run: |
+          git tag ${{ inputs.name }}
+          # push the tag to the repository and trigger the tag release workflow
+          git push origin ${{ inputs.name }}
+
+      - name: Recreate the release if it already exists
+        if: ${{ inputs.force }}
+        run: |
+          gh release delete ${{ inputs.name }} || true
+
+      - name: Create draft release
+        id: create-draft
+        run: |
+          gh release create ${{ inputs.name }} \
+            --draft \
+            --title "${{ inputs.name }}" \
+            --generate-notes \
+            --verify-tag
+
+  envs:
+    runs-on: ubuntu-latest
+    outputs:
+      build-args: ${{ steps.prepare-envs.outputs.build-args }}
+      tag: ${{ steps.create-tags.outputs.tag }}
+      otel-version: ${{ steps.load-envs.outputs.OTEL_VERSION }}
+      otel-contrib-version: ${{ steps.load-envs.outputs.OTEL_CONTRIB_VERSION }}
+      current-tag: ${{ steps.create-tags.outputs.CURRENT_TAG }}
+      previous-tag: ${{ steps.create-tags.outputs.PREVIOUS_TAG }}
+    needs:
+      - create-draft
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.name }}
 
-      # create a branch for the release and push it to the repository
-      - name: Create release branch (dry-run)
-        if: ${{ inputs.dry_run == true }}
-        run: "./hack/scripts/create_branch_and_tag.sh -v ${{ inputs.release_version }} -r origin -d"
+      - name: Prepare envs
+        id: prepare-envs
+        run: |
+          {
+          # this creates a multiline string with the envs.
```

why does this line start with lowercase? 

```suggestion
          # This creates a multiline string with the envs.
```